packages/headless/src/controllers/insight/user-actions/headless-user-actions.ts

```diff
@@ -0,0 +1,145 @@
+import {UserActionTimeline} from '../../../api/service/insight/user-actions/user-actions-response';
+import {InsightEngine} from '../../../app/insight-engine/insight-engine';
+import {configuration, insightUserActions} from '../../../app/reducers';
+import {
+  fetchUserActions,
+  incrementNumberOfSessionsAfter,
+  incrementNumberOfSessionsBefore,
+  registerUserActions,
+} from '../../../features/insight-user-actions/insight-user-actions-actions';
+import {InsightAPIErrorStatusResponse} from '../../../insight.index';
+import {
+  ConfigurationSection,
+  InsightUserActionSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {
+  buildController,
+  Controller,
+} from '../../controller/headless-controller';
+
+export interface UserActionsProps {
+  /**
+   * The options for the `UserActions` controller.
+   */
+  options: UserActionsOptions;
+}
+
+export interface UserActionsOptions {
+  /**
+   * The ticket creation date in the [ISO 8601](https://www.iso.org/iso-8601-date-and-time-format.html) format.
+   */
+  ticketCreationDate: string;
+  /**
+   * The number of sessions to include occuring before the ticket creation session.
+   * @defaultValue `0`
+   */
+  numberSessionsBefore?: number;
+  /**
+   * The number of sessions to include occuring after the ticket creation session.
+   * @defaultValue `0`
+   */
+  numberSessionsAfter?: number;
+  /**
+   * The names of custom events to exclude.
+   * @defaultValue `[]`
+   */
+  excludedCustomActions?: string[];
+}
+
+export interface UserActionsState {
+  /**
+   * The timeline of user actions.
+   */
+  timeline: UserActionTimeline;
```

Comment:
I have nothing against the current implementation , I'm just wondering it wouldn’t be more organized to have the timeline in the state as follows:
```
timeline: {
  precedingSessions: UserSession[],
  session: UserSession,
  followingSessions: UserSession
}
```

Reply:
This is precisely the kind of discussion i was hoping for! I did it this way based on the API design we have had for a while. If we want a structure like this then I would do it at the API level especially considering it isn't made yet and have the state reflect it.
If this makes the API and the state much easier to use then maybe we should do that. I'm interested in hearing others' opinions as well so let's have a longer discussion about it next Monday when Luc is back.

Reply:
I like this idea but it brings up the questions when we pass in a date, do we expect to find a session that matches that date, what ive been doing is assuming its just a date and finding the closest user actions. Also how do you guys find the date of the UA you are looking for? because in snowflake they are pretty precise (to the milliseconds)

Reply:
@mmitiche I like your suggestion to organize the sessions this way. The API will be easier to consume indeed.

Reply:
As discussed on slack we'll go with the suggested format. The PR has already been updated.